### PR TITLE
Day11 ex1 note

### PR DESCRIPTION
**Pull request by at-ducle**
#### Description (Required)
- [x] save image to sdcard
- [x] add fragment
- [x] detail fragment
#### Reference (Optional)

### CODE REVIEW

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    /**
+     * get current time
+     *
+     * @return current time
+     */
+    public String getDate() {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("EEEE MMM dd HH:mm:ss", Locale.ENGLISH);
+        String arr[] = simpleDateFormat.format(Calendar.getInstance().getTime()).split(" ");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: ERROR: Array brackets at illegal position.

File: app/src/main/java/vn/asiantech/internship/helpers/NoteOpenHelper.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                NoteDataBase.COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                NoteDataBase.COLUMN_TITLE + " TEXT NOT NULL, " +
+                NoteDataBase.COLUMN_CONTENT + " TEXT NOT NULL, " +
+                NoteDataBase.COLUMN_DATE + " TEXT NOT NULL, " +
+                NoteDataBase.COLUMN_URL_IMAGE + " TEXT);");
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS "+NoteDataBase.TABLE_NOTE);
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/helpers/NoteOpenHelper.java]: ERROR: '+' is not preceded with whitespace.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/helpers/NoteOpenHelper.java]: ERROR: '+' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package vn.asiantech.internship.interfaces;
+
+import android.app.Fragment;
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java]: ERROR: Unused import - android.app.Fragment.

File: app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * Created by ducle on 21/06/2017.
+ */
+public interface OnReplaceFragmentListener {
+    void onReplaceFragmentAdd();
+    void OnReplaceFragmentMain();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java]: ERROR: Name 'OnReplaceFragmentMain' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/ui/main/NoteActivity.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import vn.asiantech.internship.fragment.DetailNoteFragment;
+import vn.asiantech.internship.fragment.MainNoteFragment;
+import vn.asiantech.internship.interfaces.OnReplaceFragmentListener;
+import vn.asiantech.internship.ui.adapter.NoteAdapter;
+
+/**
+ * Created by ducle on 19/06/2017.
+ */
+public class NoteActivity extends AppCompatActivity implements OnReplaceFragmentListener, NoteAdapter.OnClickItemNoteListener {
+    public static final String folder = "imagenote";
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/ui/main/NoteActivity.java]: ERROR: Name 'folder' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/ui/main/NoteActivity.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        fragmentTransaction.commit();
+    }
+
+    @Override
+    public void onReplaceFragmentAdd() {
+        switchFragment(mAddNoteFragment, false, R.id.flContain);
+    }
+
+    @Override
+    public void OnReplaceFragmentMain() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/ui/main/NoteActivity.java]: ERROR: Name 'OnReplaceFragmentMain' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+                    Log.i("tag11", fullPath + getName(date) + ".png");
+                }
+                note.setTitle(mEdtTitle.getText().toString());
+                note.setContent(mEdtContent.getText().toString());
+                try {
+                    mNoteDataBase.open();
+                    mNoteDataBase.addNote(note);
+                    mNoteDataBase.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode == RESULT_OK) {
+            if (requestCode == REQUESTCODE_GALLERY) {
+                try {
+                    mBitmapImage = getImageNote(data.getData());
+                    mImageViewNote.setImageBitmap(mBitmapImage);
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+                file.createNewFile();
+            }
+
+            fOut = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.PNG, 100, fOut);
+            fOut.flush();
+            fOut.close();
+            return true;
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View view = LayoutInflater.from(container.getContext()).inflate(R.layout.fragment_detail_note, container, false);
+        initeViews(view);
+        mNoteDataBase = new NoteDataBase(getActivity());
+        try {
+            mNoteDataBase.open();
+            mNotes = mNoteDataBase.getList();
+            mNoteDataBase.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Override
+    public void onClick(View v) {
+        try {
+            mNoteDataBase.open();
+            mNoteDataBase.delete(mNote);
+            mNoteDataBase.close();
+            ((OnReplaceFragmentListener) getActivity()).OnReplaceFragmentMain();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        View view = LayoutInflater.from(container.getContext()).inflate(R.layout.fragment_main_note, container, false);
+        mRecyclerViewNote = (RecyclerView) view.findViewById(R.id.recyclerViewNote);
+        mNoteDataBase = new NoteDataBase(container.getContext());
+        try {
+            mNoteDataBase.open();
+            mNotes = mNoteDataBase.getList();
+            Log.i("TAG111", mNotes.size() + "");
+            mNoteDataBase.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    public void updateData() {
+        try {
+            mNoteDataBase.open();
+            mNotes.clear();
+            mNotes.addAll(mNoteDataBase.getList());
+            Log.i("TAG11", mNotes.size() + "");
+            mNoteAdapter.notifyDataSetChanged();
+            mNoteDataBase.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java (hunk 3 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.app.Fragment'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java (hunk 4 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/interfaces/OnReplaceFragmentListener.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+                    Log.i("tag11", fullPath + getName(date) + ".png");
+                }
+                note.setTitle(mEdtTitle.getText().toString());
+                note.setContent(mEdtContent.getText().toString());
+                try {
+                    mNoteDataBase.open();
+                    mNoteDataBase.addNote(note);
+                    mNoteDataBase.close();
+                } catch (IOException e) {
+                    Log.d("tag","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode == RESULT_OK) {
+            if (requestCode == REQUESTCODE_GALLERY) {
+                try {
+                    mBitmapImage = getImageNote(data.getData());
+                    mImageViewNote.setImageBitmap(mBitmapImage);
+                } catch (Exception e) {
+                    Log.i("tag111","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+                file.createNewFile();
+            }
+
+            fOut = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.PNG, 100, fOut);
+            fOut.flush();
+            fOut.close();
+            return true;
+        } catch (IOException e) {
+            Log.d("tag1111","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View view = LayoutInflater.from(container.getContext()).inflate(R.layout.fragment_detail_note, container, false);
+        initeViews(view);
+        mNoteDataBase = new NoteDataBase(getActivity());
+        try {
+            mNoteDataBase.open();
+            mNotes = mNoteDataBase.getList();
+            mNoteDataBase.close();
+        } catch (IOException e) {
+            Log.d("tag","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Override
+    public void onClick(View v) {
+        try {
+            mNoteDataBase.open();
+            mNoteDataBase.delete(mNote);
+            mNoteDataBase.close();
+            ((OnReplaceFragmentListener) getActivity()).onReplaceFragmentMain();
+        } catch (IOException e) {
+            Log.d("tag1","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/DetailNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+        View view = LayoutInflater.from(container.getContext()).inflate(R.layout.fragment_main_note, container, false);
+        mRecyclerViewNote = (RecyclerView) view.findViewById(R.id.recyclerViewNote);
+        mNoteDataBase = new NoteDataBase(container.getContext());
+        try {
+            mNoteDataBase.open();
+            mNotes = mNoteDataBase.getList();
+            Log.i("TAG111", mNotes.size() + "");
+            mNoteDataBase.close();
+        } catch (IOException e) {
+            Log.d("tag","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+    public void updateData() {
+        try {
+            mNoteDataBase.open();
+            mNotes.clear();
+            mNotes.addAll(mNoteDataBase.getList());
+            Log.i("TAG11", mNotes.size() + "");
+            mNoteAdapter.notifyDataSetChanged();
+            mNoteDataBase.close();
+        } catch (IOException e) {
+            Log.d("tag1","ERROR");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/MainNoteFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java (hunk 14 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/fragment/AddNoteFragment.java]: WARNING: [AvoidDuplicateLiterals] The String literal "ERROR" appears 4 times in this file; the first occurrence is on line 100
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Comment by at-tamnguyen2** [comment]
@at-ducle khi bấm save note nó bị crash app, em check lại giúp anh. Thanks

**Comment by at-tamnguyen2** [comment]
@at-ducle 
LGTM
Thank you^^

**Comment by at-tienhoang** [comment]
LGTM, thanks!

**Pull request merged**